### DISCUSSION TITLE
Problems with authWithOAuth2Code in SvelteKit

### DISCUSSION THREAD

**Discussion by dimitarnikolovv** [pocketbase/pocketbase#3849] [Q&A]
I am facing a bit of an issue trying to update the `authStore` while using `authWithOAuth2Code`. 
I have a form action that sets 2 cookies, one for the state and one for the verifier that the `authProvider` returns.

```js
oauth2: async ({ cookies, locals, url }) => {
    const authMethods = await locals.pb.collection('users').listAuthMethods();

    const redirectURL = `${url.origin}/api/oauth2`;
    const googleAuth = authMethods.authProviders[0];
    const authProviderRedirect = `${googleAuth.authUrl}${redirectURL}`;

    const state = googleAuth.state;
    const verifier = googleAuth.codeVerifier;

    cookies.set('state', state);
    cookies.set('verifier', verifier);

    throw redirect(302, authProviderRedirect);
  }
```

Then I am getting these cookies on the server side and I am calling `authWithOAuth2Code` and it works. It sets the `authStore` and everything, the problem is that when I throw a redirect the `authStore` gets overridden most likely by the handle hook. 

```js
try {
    await locals.pb
      .collection('users')
      .authWithOAuth2Code(provider.name, code, expectedVerifier, redirectURL, {
        firstName: '',
        lastName: '',
        age: 1,
        gender: 'other',
      });
  } catch (err) {
    console.log(err);
  }

  throw redirect(303, '/');  //bad things happen here
```

hooks.server.js
```js
export const handle = async ({ event, resolve }) => {
  event.locals.pb = pb;
  event.locals.pb.authStore.loadFromCookie(event.request.headers.get('cookie') || '');

  try {
    if (event.locals.pb.authStore.isValid) {
      await event.locals.pb.collection('users').authRefresh({ expand: 'allowedToRate' });
      event.locals.user = serializeNonPOJOs(event.locals.pb.authStore.model);
    }
  } catch (_) {
    event.locals.pb.authStore.clear();
    event.locals.user = undefined;
  }

  const response = await resolve(event);

  console.log(event.locals.pb.authStore);

  response.headers.append('set-cookie', event.locals.pb.authStore.exportToCookie({
      secure: dev ? false : true,
      httpOnly: true,
    })
  );

  return response;
};
``` 

I am curious if someone had a similar behavior and if there is a way to fix it. Or maybe the problem lies somewhere else. It's most likely a SvelteKit thing though.

**Comment by ganigeorgiev** [comment]
I believe this is the same as https://github.com/pocketbase/pocketbase/discussions/903#discussioncomment-4369424. Try to export the cookie with `sameSite: 'lax'`, but make sure to check the difference between `Strict` (for example in https://web.dev/articles/samesite-cookies-explained).

**Reply by dimitarnikolovv** [reply]
`Lax` worked, but I found out that it was a problem with some cookies caching because even though I was setting a new cookie, in the network tab there was a much older cookie every time. After clearing the cookies it works using `Strict`.
